lib/primer/dsl/view_helper.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Primer
+  # :nodoc:
+  module DSL
+    # DSL to allow components to register a View Helper for shorthand calls.
+    #
+    # Example:
+    #
+    # class MyComponent < ViewComponent::Base
+    #   include Primer::DSL::ViewHelper
+    #   view_helper :my_component
+    # end
+    module ViewHelper
+      extend ActiveSupport::Concern
+
+      class ViewHelperAlreadyDefined < StandardError; end
+
+      included do
+        class_attribute :helpers, instance_writer: false, default: {}
```

Comment:
```suggestion
        class_attribute :primer_helpers, instance_writer: false, default: {}
```

Super minor, but thoughts on "namespacing" the attribute to help avoid conflicts with other libraries/gems?